### PR TITLE
[java] handle `getNamedCookie` and `deleteNamedCookie` for empty strings

### LINKED ISSUE #15044: [🚀 Feature]: Use dedicated WebDriver Spec endpoint to get named cookie
### Feature and motivation

**Problem:** To get cookie by name we fetch all cookies and then filter them out internally.

**Spec**: https://www.w3.org/TR/webdriver2/#get-named-cookie

### Usage example

C#:
```
driver.Manage().Cookies.GetCookieNamed("abc");
```

**Solution**

This method should send GET `/session/{sessionId}/cookie/{name}` request. `{name}` is not null or empty, otherwise remote end returns the list of all cookies.

### PR DESCRIPTION
**Pull request by Delta456**
### **User description**
**Thanks for contributing to Selenium!**
**A PR well described will help maintainers to quickly review and merge it**

Before submitting your PR, please check our [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) guidelines.
Avoid large PRs, help reviewers by making them as simple and short as possible.

### Description

Fixes https://github.com/SeleniumHQ/selenium/issues/15044 for Java bindings

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [x] I have read the [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

___

### **PR Type**
Bug fix, Tests

___

### **Description**
- Added validation to prevent empty or null cookie names in `deleteCookieNamed` and `getCookieNamed` methods.

- Introduced unit tests to verify exceptions for empty or whitespace cookie names.

- Enhanced error handling for cookie management in `RemoteWebDriver`.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Bug fix</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>RemoteWebDriver.java</strong><dd><code>Add validation for empty cookie names in RemoteWebDriver</code>&nbsp; </dd></summary>
<hr>

java/src/org/openqa/selenium/remote/RemoteWebDriver.java

<li>Added validation to check for empty or whitespace cookie names in <br><code>deleteCookieNamed</code>.<br> <li> Added validation to check for empty or whitespace cookie names in <br><code>getCookieNamed</code>.<br> <li> Enhanced error handling for cookie management methods.

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/15092/files#diff-0cf141b01e48a733ec9037c51d8e610e7727d05a6b79aa0176def7c3a0fc311b">+6/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>CookieImplementationTest.java</strong><dd><code>Add tests for empty cookie name validation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

java/test/org/openqa/selenium/CookieImplementationTest.java

<li>Added tests to verify exceptions for empty or whitespace cookie names <br>in <code>deleteCookieNamed</code>.<br> <li> Added tests to verify exceptions for empty or whitespace cookie names <br>in <code>getCookieNamed</code>.<br> <li> Ensured robust test coverage for cookie management.

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/15092/files#diff-61896345e2b54f1347497c98ed2a822a59a7b2a132d6dffa7d8922c60caa56b9">+13/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

**Review by VietND96 — APPROVED** [review]

**Review by VietND96 — CHANGES_REQUESTED** [review]
CI tests are failing. That needs to be fixed.

**Review by VietND96 — APPROVED** [review]

**Comment by VietND96** [comment]
@diemol, @pujagani, can you double check once?

**Comment by pujagani** [comment]
The failing tests are not associated with the changes made here

File: java/src/org/openqa/selenium/remote/RemoteWebDriver.java [hunk 1]
```diff
@@ -927,6 +930,9 @@ public Set<Cookie> getCookies() {
 
     @Override
     public Cookie getCookieNamed(String name) {
+      if (name == null || name.isBlank()) {
```
Comment (pujagani) [inline java/src/org/openqa/selenium/remote/RemoteWebDriver.java:933]: To fix the underlying issue, we need to add the endpoint "/session/sessionId/cookie/name" and use that to get the cookie

Comment (Delta456) [reply java/src/org/openqa/selenium/remote/RemoteWebDriver.java:933]: I see `defineCommand(GET_COOKIE, get(cookie + "/:name"));` in `java/src/org/openqa/selenium/remote/codec/AbstractHttpCommandCodec.java`

**Review by asolntsev — APPROVED** [review]

**Review by pujagani — APPROVED** [review]
Thank you! LGTM
